### PR TITLE
Renamings in `contiguity.{cpp,h}`

### PR DESCRIPTION
**Pull request by zasdfgbnm**
Split out from https://github.com/NVIDIA/Fuser/pull/276, mostly `root` -> `allocation`

### CODE REVIEW

**Review by naoyam — APPROVED** [review]
LGTM

File: csrc/index_compute.cpp [hunk 1]
```diff
@@ -654,7 +654,6 @@ IndexCompute::IndexCompute(
       zero_domains_(std::move(zero_domains)),
       zero_merged_in_(std::move(zero_merged_in)),
       contig_ids_{contig_finder.contigIDs()},
-      root_to_indexed_id_{contig_finder.rootToIndexedID()},
```
Comment (naoyam) [inline csrc/index_compute.cpp:657]: Why was this removed?

Comment (zasdfgbnm) [reply csrc/index_compute.cpp:657]: This is never used, `root_to_indexed_id_` is also removed.
